OnServiceChange/handler.ts

```diff
@@ -243,27 +239,39 @@ export const createServiceChangeHandler = (
       context.log(`${logPrefix}|Received ${d.length} documents.`);
       return d;
     },
-    RA.map(validateDocument),
+    // Validate each document
+    RA.map(
+      flow(
+        validateDocument,
+        E.mapLeft(err => {
+          const reason = readableReport(err);
+          trackIgnoredInvalidIncomingDocument(telemetryClient)(
+            document,
+            reason
+          );
```

Comment:
as `validateDocument` does not hide validation details, we can enrich monitoring to ease eventual troubleshooting

Reply:
What do you mean? I see we're tracking `reason` as a `readableReport(err)`, where `err` is the validation details

Reply:
`validateDocument` is actually an alias on `RetrievedService.decode`: https://github.com/pagopa/io-subscription-migration/blob/7f58f2f3cf7aa4a151f574db9191bad4399f4e8e/OnServiceChange/handler.ts#L47

Reply:
Ok, I see you updated the code after this comment, sorry!